actors/builtin/multisig/multisig_actor.go

```diff
@@ -423,21 +417,35 @@ func (a Actor) approveTransaction(rt vmr.Runtime, txnID TxnID, proposalHash []by
 }
 
 func isSigner(rt vmr.Runtime, st *State, address addr.Address) bool {
+	candidateIdAddr := address
+	if candidateIdAddr.Protocol() != addr.ID {
+		idAddr, found := rt.ResolveAddress(candidateIdAddr)
+		if found {
+			candidateIdAddr = idAddr
+		}
+	}
+
 	for i, ap := range st.Signers {
 		if address == ap {
 			return true
 		}
 
-		if ap.Protocol() != addr.ID {
+		switch ap.Protocol() {
+		case addr.ID:
+			if candidateIdAddr == ap {
+				return true
+			}
+		default:
 			idAddr, found := rt.ResolveAddress(ap)
 			if found {
 				st.Signers[i] = idAddr
-				if idAddr == address {
+				if idAddr == candidateIdAddr {
```

Comment:
I think this is correct, but it's hard to mentally check because there are three different places where the addresses are compared. I think it would be simpler for the algo to be:

- candidate = resolve(candidate)
- for each signer:
  - signer = resolve(signer)
  - if candidate == signer: return true

... where `resolve()` include the checking the protocol etc as it does already here.